wfe/wfe.go

```diff
@@ -1154,6 +1165,12 @@ func (wfe *WebFrontEndImpl) Registration(ctx context.Context, logEvent *requestE
 	// the key of the updated registration object is going to be the same as the
 	// key of the current one, so we set it here. This ensures we can cleanly
 	// serialize the update as JSON to send via AMQP to the RA.
+	if features.Enabled(features.AllowKeyRollover) && update.Key != emptyKey && update.Key != currReg.Key {
```

Comment:
The line bellow this `if` statement always overwrites the key in `update` with the key from `currReg`. The feature flagging is to just maintain the current behavior until we flip the flag, after which we'll want to tell people how they're actually supposed to change their key.